magicctapipe/scripts/lst1_magic/lst1_magic_dl1_stereo_to_dl2.py

```diff
@@ -2,82 +2,266 @@
+            )
+
+            event_coord_per_flip = event_coord_per_flip.altaz
+
+            df_altaz_per_flip = pd.DataFrame(
+                data={
+                    "reco_alt": event_coord_per_flip.alt.to_value(u.deg),
+                    "reco_az": event_coord_per_flip.az.to_value(u.deg),
+                    "flip": flip,
+                },
+                index=df_events.index,
+            )
+
+            reco_params_flips = reco_params_flips.append(df_altaz_per_flip)
+
+    reco_params_flips.set_index("flip", append=True, inplace=True)
+    reco_params_flips.sort_index(inplace=True)
+
+    group_size = reco_params_flips.groupby(["obs_id", "event_id"]).size()
+    reco_params_flips["multiplicity"] = group_size
+
+    # Then, we get the flip combination minimizing the sum of the
+    # angular distances between the head and tail candidates per shower
+    # event. In order to speed up the calculations, here we separate the
+    # input events by the telescope combination types.
+
+    reco_params = pd.DataFrame()
+
+    for tel_ids in TEL_COMBINATIONS.values():
+
+        multiplicity = 2 * len(tel_ids)
+
+        df_events = reco_params_flips.query(
+            f"(multiplicity == {multiplicity}) & (tel_id == {tel_ids})"
+        ).copy()
+
+        df_events["multiplicity"] = df_events.groupby(["obs_id", "event_id"]).size()
+        df_events.query(f"multiplicity == {multiplicity}", inplace=True)
+
+        n_events = len(df_events.groupby(["obs_id", "event_id"]).size())
+
+        # Here we first define all the possible flip combinations. For
+        # example, in case that we have two telescope images, in total
+        # 4 combinations can be defined as follows:
+        #   [(head, head), (head, tail), (tail, head), (tail, tail)]
+        # where the i-th element of each tuple means the i-th telescope
+        # image. In case of 3 images we have in total 8 combinations.
+
+        flip_combinations = np.array(
+            list(itertools.product([0, 1], repeat=len(tel_ids)))
+        )
+
+        # Next, we define all the possible any 2 telescope combinations.
+        # For example, in case of 3 telescopes, in total 3 combinations
+        # are defined as follows:
+        #                 [(1, 2), (1, 3), (2, 3)]
+        # where the elements of the tuples mean the telescope IDs. In
+        # case of 2 telescopes there is only one combination.
+
+        tel_any2_combinations = list(itertools.combinations(tel_ids, 2))
+
+        distances = np.zeros((len(flip_combinations), n_events))
+
+        for i_flip, flip_combo in enumerate(flip_combinations):
+
+            container = {}
+
+            # Set the directions of a given flip combination
+            for tel_id, flip in zip(tel_ids, flip_combo):
+                container[tel_id] = df_events.query(
+                    f"(tel_id == {tel_id}) & (flip == {flip})"
+                )
+
+            for tel_any2_combo in tel_any2_combinations:
```

Comment:
wouldn't this work instead:
```
for tel_id_1,tel_id_2 in tel_any2_combinations:
```

Reply:
It does not work, because the `tel_any2_combinations` is a list containing tuples of any 2 tels combination, so we have to loop with one variable. In case of `tel_id = [1, 2, 3]`, the `tel_any2_combinations` becomes `[(1, 2), (1, 3), (2, 3)]`, so the `tel_any2_combo` will be `(1, 2)`, `(1, 3)` or `(2, 3)`.

Reply:
hmm, are you sure?
```
In [3]: tel_any2_combinations=[(1,2),(1,3),(2,3)]

In [4]: for tel_id_1,tel_id_2 in tel_any2_combinations:
   ...:     print(tel_id_1, tel_id_2)
1 2
1 3
2 3

In [5]: for (tel_id_1,tel_id_2) in tel_any2_combinations:
   ...:     print(tel_id_1, tel_id_2)
1 2
1 3
2 3
```

Reply:
Oh really?? OK then sorry, it's my bad 😅. I will implement your suggestion.